src/runtime/extensiontype.cs

```diff
@@ -89,8 +94,22 @@ public static void tp_dealloc(IntPtr ob)
         {
             // Clean up a Python instance of this extension type. This
             // frees the allocated Python object and decrefs the type.
-            ManagedType self = GetManagedObject(ob);
-            FinalizeObject(self);
+            var self = (ExtensionType)GetManagedObject(ob);
+            self.Dealloc();
+        }
+
+        protected override void OnSave()
+        {
+            base.OnSave();
+            Runtime.XIncref(pyHandle);
+        }
+
+        protected override void OnLoad()
+        {
+            base.OnLoad();
+            GCHandle gc = AllocGCHandle(TrackTypes.Extension);
+            Marshal.WriteIntPtr(pyHandle, ObjectOffset.magic(tpHandle), (IntPtr)gc);
+            Runtime.PyObject_GC_UnTrack(pyHandle);
```

Comment:
huh... it already comment at `ExtensionType` constructor, needs a repeat?
```
            // We have to support gc because the type machinery makes it very
            // hard not to - but we really don't have a need for it in most
            // concrete extension types, so untrack the object to save calls
            // from Python into the managed runtime that are pure overhead.
```
Since the mechanism of soft-shutdown, it turns back to GC, after the object reloaded, it just needs the same operation.